Sources/CoreMetrics/SystemMetrics.swift

```diff
@@ -0,0 +1,172 @@
+    var residentMemory: Int32? { get }
+    var startTimeSeconds: Int32? { get }
+    var cpuSeconds: Int32? { get }
+    var maxFds: Int32? { get }
+    var openFds: Int32? { get }
+}
+
+#if os(Linux)
+private struct LinuxSystemMetricsProvider: SystemMetrics {
+    let virtualMemory: Int32?
+    let residentMemory: Int32?
+    let startTimeSeconds: Int32?
+    let cpuSeconds: Int32?
+    let maxFds: Int32?
+    let openFds: Int32?
+
+    private enum SystemMetricsError: Error {
+        case FileNotFound
+    }
+
+    init(pid: String) {
+        let ticks = Int32(_SC_CLK_TCK)
+        do {
+            guard let statString =
+                try String(contentsOfFile: "/proc/\(pid)/stat", encoding: .utf8)
+                .split(separator: ")")
+                .last
+            else { throw SystemMetricsError.FileNotFound }
+            let stats = String(statString)
+                .split(separator: " ")
+                .map(String.init)
+            self.virtualMemory = Int32(stats[safe: 20])
+            if let rss = Int32(stats[safe: 21]) {
+                self.residentMemory = rss * Int32(_SC_PAGESIZE)
+            } else {
+                self.residentMemory = nil
+            }
+            if let startTimeTicks = Int32(stats[safe: 19]) {
+                self.startTimeSeconds = startTimeTicks / ticks
+            } else {
+                self.startTimeSeconds = nil
+            }
+            if let utimeTicks = Int32(stats[safe: 11]), let stimeTicks = Int32(stats[safe: 12]) {
+                let utime = utimeTicks / ticks, stime = stimeTicks / ticks
+                self.cpuSeconds = utime + stime
+            } else {
+                self.cpuSeconds = nil
+            }
+        } catch {
+            self.virtualMemory = nil
+            self.residentMemory = nil
+            self.startTimeSeconds = nil
+            self.cpuSeconds = nil
+        }
+        do {
+            guard
+                let line = try String(contentsOfFile: "/proc/\(pid)/limits", encoding: .utf8)
+                .split(separator: "\n")
+                .first(where: { $0.starts(with: "Max open file") })
+                .map(String.init) else { throw SystemMetricsError.FileNotFound }
+            self.maxFds = Int32(line.split(separator: " ").map(String.init)[safe: 3])
+        } catch {
+            self.maxFds = nil
+        }
+        do {
+            let fm = FileManager.default,
+                items = try fm.contentsOfDirectory(atPath: "/proc/\(pid)/fd")
+            self.openFds = Int32(items.count)
+        } catch {
+            self.openFds = nil
+        }
+    }
+}
+
+#else
+private struct UnsuportedMetricsProvider: SystemMetrics {
+    let virtualMemory: Int32?
+    let residentMemory: Int32?
+    let startTimeSeconds: Int32?
+    let cpuSeconds: Int32?
+    let maxFds: Int32?
+    let openFds: Int32?
+
+    init(pid: String) {
+        #warning("System Metrics are not implemented on non-Linux platforms yet.")
```

Comment:
So this worked ok while we hide this type on non linux, but I don't think we should hide it, so this line has to be removed in order to not cause compilation warnings